Services/PostService.ts

```diff
@@ -0,0 +1,34 @@
+import { SearchPostDto } from '../Dto/SearchPostResponseDto';
+import { Post } from '../Models/Post';
+import { User } from '../Models/User';
+import { request } from '../Utils/RequestUtils';
+
+export class PostService {
+  private searchInStr(text: string, term: string): boolean {
+    return text.toLowerCase().split(' ').includes(term.toLowerCase());
+  }
+
+  public async searchTerm(term: string): Promise<SearchPostDto> {
+    const postsData = await request<Post[]>('/posts');
+    const post = postsData?.find((p) => {
```

Comment:
Let's move all of this to a util so that if we want to do unit tests, we can test it easily without having to mock.